config/initializers/attribute_types.rb

```diff
@@ -1,9 +1,9 @@
 class Gbp < ActiveModel::Type::Decimal
   def cast(value)
-    if value.is_a?(Numeric) || value.blank? || value.gsub(/[0-9,.£]/, "").empty?
+    if value.is_a?(Numeric) || value.blank? || value.gsub(/[0-9,.£]+\s?\z/, "").empty?
```

Comment:
This will only work for a single space at the end of the string. Multiple spaces, or tabs, or whitespace at the beginning of the string, will suffer the same problem. So how about instead you do:

```ruby
value.is_a?(Numeric) || value.blank? || value.gsub(/[0-9,.£]/, "").blank?
```

Reply:
this lets you enter `123 000`, which is accepted but is converted to `123` so we get inaccurate results. We(ruby?) are already removing whitespace at the start of any value that is entered.

Reply:
i have amended the regex to remove all trailing whitespace instead

Reply:
If you enter `  £20000` as it stands that will still cause a string rather than a decimal to be stored in the session, leading to a CFE crash. So I think we really do need to deal with leading whitespace.  Maybe it's `value.strip.gsub(/[0-9,.£]/, "").empty?`, because that will return true for leading and trailing whitespace but false for middle-of-string whitespace.